src/extensions/default/InlineColorEditor/main.js

```diff
@@ -59,7 +67,45 @@ define(function (require, exports, module) {
         } while (match && (pos.ch < start || pos.ch > end));
 
         if (!match) {
-            return null;
+            // Check if the cursorLine has a CSS rule of type color
+            var cssPropertyName, semiColonPos, colonPos, colorValue, cursorLineSubstring, firstCharacterPos;
+
+            // Get the css property name after removing spaces and ":" so that we can check for it in the file ColorProperties.json
+            cssPropertyName = cursorLine.split(':')[0].trim();
+
+            if (!cssPropertyName || !properties[cssPropertyName]) {
+                return null;
+            }
+
+            if (properties[cssPropertyName].type === "color") {
+                colonPos = cursorLine.indexOf(":");
+                semiColonPos = cursorLine.indexOf(";");
+                if (semiColonPos !== -1) {
+                    // edit the color value of an existing css rule
```

Comment:
@hkirat thanks for pointing out the possibility of `color : ;`. However, I do need that code.
Reason: The previous code allows the user to change the color value only if the cursor is on the color value. It doesn't allow the user to use the color picker by pressing ctrl + e anywhere on the line of a CSS rule of type color and change the color value. We need that functionality.